### ISSUE TITLE
FileNotFound exceptions on connecting Chlorine to Leiningen nREPL on Windows

### ISSUE THREAD

**Issue by SarmBoJim** [mauricioszabo/atom-chlorine#191]
Hi again,
Sorry if this is not a bug or the right way to contact regarding this issue, I am fairly new. Any help with this would be great, if it's not too much hassle.
When I open a socket REPL with Leiningen from my project's root directory and then connect Chlorine to it, the Chlorine REPL tab displays the following messages:

"Execution error (FileNotFoundException) at inventory.core/eval1560 (form-init18339748147166039704.clj:1).
Could not locate orchard/info__init.class, orchard/info.clj or orchard/info.cljc on classpath.
Execution error (FileNotFoundException) at inventory.core/eval1562 (form-init18339748147166039704.clj:1).
Could not locate orchard/xref__init.class, orchard/xref.clj or orchard/xref.cljc on classpath.
Execution error (FileNotFoundException) at inventory.core/eval1564 (form-init18339748147166039704.clj:1).
Could not locate orchard/clojuredocs__init.class, orchard/clojuredocs.clj or orchard/clojuredocs.cljc on classpath."

---

The error happens when connecting to nREPL ports. Does not affect the behavior of Chlorine in any way, and the error only happens once.

**Comment by mauricioszabo** [comment]
Seems like a bug. Does it connects, works, and only prints this on the REPL, or does it affect usage of plug-in in any way?

**Comment by SarmBoJim** [comment]
It connects successfully yes, it prints those exceptions on the REPL, I worked through most of a chapter on Java interop last night with it so it seems to be working.
There is a bit of weirdness with the keymappings though, the ones it gives me for Chlorine commands seem to be a bit different from the ones I pasted into Atom's keymaps.cson, maybe I am just doing something wrong though, I'll investigate that a bit further and get back to you with some more thorough info.
I'll get some screenshots to you also.

**Comment by SarmBoJim** [comment]
OK I've looked into things a bit more and taken some screens.
My procedure is I switch to my project root directory in Windows command prompt, execute the command "lein repl" then grab the resulting port number, start Atom, (my project folder is already loaded in Atom), execute the command "Connect Socket REPL" from the command palette which opens the Chlorine REPL tab, a message "Clojure nREPL Connected" pops up and the Chlorine REPL tab is immediately filled with the FileNotFound exceptions as shown in the following screenshot:

![Exceptions on connect](https://user-images.githubusercontent.com/53518185/83080262-de83a580-a0d1-11ea-8f63-26ce22b18762.jpg)

Then if I start writing a function call in my Clojure source file, as soon as I type the last character of the function name then one more FileNotFound exception gets added to the list for compliment as shown in the following screen:

![Additional exception on first function name](https://user-images.githubusercontent.com/53518185/83080479-69fd3680-a0d2-11ea-9af8-836102d65e5e.jpg)

That happens whether I type foo as a function name or something else like doc and the exception only appears as soon as I type the last character of the function name. It is only for the first function name I type after connecting Chlorine.  Subsequent function calls that I type out do not produce any more exceptions. So after that first function name being typed, everything seems to go smoothly, things work and output is produced as in the following screen:

![First outputs](https://user-images.githubusercontent.com/53518185/83081107-007e2780-a0d4-11ea-96de-2411c0c288b3.jpg)

Not sure if this has anything to do with it, it's been many years since I did my Networking paper at uni so don't understand network stuff and ports very much - the port number I specify in my project.clj file is different from the one I receive when starting the REPL with Leiningen on the command prompt. Leiningen always gives me a different port number.
Here is my project.clj config with port number:

![Project config file and socket configuration](https://user-images.githubusercontent.com/53518185/83081494-f3156d00-a0d4-11ea-8d6e-853ed8ebdb7b.jpg)

And here is the port number I receive from Leiningen (though as I say, it changes every time I start a new REPL):

![REPL server and port on command prompt](https://user-images.githubusercontent.com/53518185/83081588-2c4ddd00-a0d5-11ea-8641-90b4f00625c3.jpg)

The keymappings issue I mentioned in my last post isn't as weird as I thought at first and might just be a misunderstanding on my behalf of how the keymappings work in Atom, I'll just post that as a separate issue as I think it is unconnected to this and you can just close it if it is not a bug and rather my ignorance.

**Comment by seancorfield** [comment]
@SarmBoJim The port number in your JVM options is the Socket REPL Server port number -- you can connect Chlorine to that.  This is what Chlorine was designed for -- working with Socket REPL Servers.

The port number that Leiningen reports in the nREPL Server that Leiningen starts. As you've seen, you can also connect Chlorine to that. Chlorine's support for nREPL is a recent addition and I suspect the Orchard-related errors are either because your project doesn't have some development dependencies that nREPL/Orchard requires for auto-completion etc, or because of a version mismatch (Chlorine may expect more up-to-date versions of some of that?).

For now, ignore the port Leiningen reports -- connect Chlorine to port 5555 per your JVM options.

**Comment by SarmBoJim** [comment]
@seancorfield Ah OK, thank you very much for the clarification. However I get a different issue when I try to connect to the port number 5555 as in my project.clj jvm options. It won't connect. I get the following message pop up:

![Connection refused](https://user-images.githubusercontent.com/53518185/83096858-d8ee8580-a0fa-11ea-898d-e8f4d8b0f656.jpg)

Could it be Windows Firewall is blocking the port? Or maybe that port number 5555 is reserved by Windows? I tried using 8086 in my project.clj and it refused that connection too. 
Thinking maybe this is not the best place to be asking for help in case this is not a bug and just my ignorance as I am beginning to suspect, so what would be the best place to ask for Chlorine help?

**Comment by seancorfield** [comment]
The Clojurians Slack is the best place for interactive help: https://clojurians.slack.com with auto-signup at http://clojurians.net -- there's a #chlorine channel there.

Happy to help you debug this more interactively once you're on Slack. I'll leave it up to @mauricioszabo to decide whether this should be closed...

**Comment by SarmBoJim** [comment]
OK thank you very much, I'll join up on Slack :)

**Comment by SarmBoJim** [comment]
@mauricioszabo  OK everything's working for me with Chlorine now thank you very much, my post on the #chlorine Slack channel clarifies my stupidity, sorry for the waste of time.
Those FileNotFound exceptions when connecting Chlorine to the nREPL could still be a bug though?
Anyway, thank you very much for the help and for the awesome Chlorine!

**Comment by mauricioszabo** [comment]
@SarmBoJim well, it's still a bug. The problem is that `nREPL` prints (by itself, it appears) when some exception happens.

Chlorine, when you connect, `require`s lots of files to see if they are present. If they're not, it does not enable commands for these namespaces. The list of projects it requires is `compliment`, `orchard`, and `clj-suitable` (if you're using ClojureScript). After it requires for the first time, it'll cache the results (that's the reason you're only seeing this issue only once).

I'll keep this bug open, but I'll add more info about it so we can debug then and fix nREPL connections. Thanks for reporting and for taking your time to document every step :+1:

**Comment by SarmBoJim** [comment]
You're welcome and thank you, I'm glad at least some of it was helpful :)

**Comment by mauricioszabo** [comment]
Fixed on 0.7.5
